src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/ThreadStaticLifetimeScope.cs

```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#if !FEATURE_REMOTING && !FEATURE_ASYNCLOCAL
 namespace Castle.MicroKernel.Lifestyle.Scoped
 {
-#if SILVERLIGHT
```

Comment:
I just realised we can actually just delete `ThreadStaticLifetimeScope` since it was only used on Silverlight, it'll also simplify some of the preprocessor conditionals in other files.